src/Differ.php

```diff
@@ -69,50 +66,69 @@ private function validateDiffInput($input): string
     }
 
     /**
-     * Takes input of the diff array and returns the old array.
-     * Iterates through diff line by line,
+     * Takes input of the diff array and returns the common parts.
+     * Iterates through diff line by line.
      *
      * @param array $diff
+     * @param int   $lineThreshold
      *
      * @return array
      */
-    private function checkIfDiffInOld(array $diff): array
+    private function getCommonChunks(array $diff, int $lineThreshold = 5): array
     {
-        $inOld = false;
-        $i     = 0;
-        $old   = [];
-
-        foreach ($diff as $line) {
-            if ($line[1] === 0 /* OLD */) {
-                if ($inOld === false) {
-                    $inOld = $i;
+        $diffSize     = \count($diff);
+        $capturing    = false;
+        $chunkStart   = 0;
+        $chunkSize    = 0;
+        $commonChunks = [];
+
+        for ($i = 0; $i < $diffSize; ++$i) {
+            if ($diff[$i][1] === 0 /* OLD */) {
+                if ($capturing === false) {
+                    $capturing  = true;
+                    $chunkStart = $i;
+                    $chunkSize  = 0;
+                } else {
+                    ++$chunkSize;
                 }
-            } elseif ($inOld !== false) {
-                if (($i - $inOld) > 5) {
-                    $old[$inOld] = $i - 1;
+            } elseif ($capturing !== false) {
+                if ($chunkSize >= $lineThreshold) {
+                    $commonChunks[$chunkStart] = $chunkStart + $chunkSize;
                 }
 
-                $inOld = false;
+                $capturing = false;
             }
+        }
 
-            ++$i;
+        if ($capturing !== false && $chunkSize >= $lineThreshold) {
+            $commonChunks[$chunkStart] = $chunkStart + $chunkSize;
         }
 
-        return $old;
+        return $commonChunks;
     }
 
     /**
      * Generates buffer in string format, returning the patch.
      *
      * @param array $diff
-     * @param array $old
-     * @param int   $start
-     * @param int   $end
      *
      * @return string
      */
-    private function getBuffer(array $diff, array $old, int $start, int $end): string
+    private function getDiff(array $diff): string
     {
+        $old   = $this->getCommonChunks($diff, 5);
+        $start = isset($old[0]) ? $old[0] : 0;
+        $end   = \count($diff);
+
+        if (\count($old)) {
```

Comment:
This is what the removed dead code of https://github.com/sebastianbergmann/diff/pull/51 was supposed to do.